test/test_datasets.py

```diff
@@ -832,11 +846,27 @@ def test_slice_error(self):
             with pytest.raises(ValueError, match="Index must be of type integer"):
                 dataset[:2]
 
+    def test_segmentation_kind(self):
```

Comment:
Really not a fan of that test's logic, but I don't know how to better test that. @pmeier  LMK if there are better options?